ide/che-core-ide-generators/src/main/java/org/eclipse/che/util/CompilingGwtXmlGenerator.java

```diff
@@ -0,0 +1,225 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.util;
+
+import com.google.common.collect.ImmutableSet;
+
+import org.apache.commons.io.FileUtils;
+import org.reflections.Reflections;
+import org.reflections.scanners.ResourcesScanner;
+import org.reflections.util.ConfigurationBuilder;
+import org.reflections.util.FilterBuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+import static org.eclipse.che.util.IgnoreUnExistedResourcesReflectionConfigurationBuilder.getConfigurationBuilder;
+
+/**
+ * Generates uber IDE.gwt.xml from part found in class path.
+ * @author Sergii Kabashniuk
+ */
+public class CompilingGwtXmlGenerator {
+
+    public static final String DEFAULT_GWT_XML_PATH    = "org/eclipse/che/ide/IDE.gwt.xml";
+    public static final String DEFAULT_GWT_ETNRY_POINT = "org.eclipse.che.ide.client.IDE";
+    public static final String DEFAULT_STYLE_SHEET     = "IDE.css";
+
+    private final GwtXmlGeneratorConfig config;
+
+    public CompilingGwtXmlGenerator(GwtXmlGeneratorConfig config) {
+        this.config = config;
+    }
+
+    public File generateGwtXml() throws IOException {
+        File gwtXml = new File(config.getGenerationRoot(), config.getGwtFileName());
+        if (gwtXml.isDirectory() || gwtXml.exists()) {
+            throw new IOException(gwtXml.getAbsolutePath() + " already exists or directory");
+        }
+        StringBuilder builder = new StringBuilder();
+        builder.append("<?xml version=\"1.0\" encoding=\"UTF-8\"?>").append("\n");
```

Comment:
we should use templating with something like org.antlr.ST4

here is an example
https://github.com/eclipse/che/blob/master/core/che-core-typescript-dto-maven-plugin/src/main/resources/org/eclipse/che/plugin/typescript/dto/typescript.template

we have so many builder.append there that it's quite impossible to read